lib/presentation/screens/details_screen/details_screen.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:marvelapp_flutter/data/repository/dio_marvell_repository.dart';
+import 'package:marvelapp_flutter/data/repository/marvell_repository.dart';
+import 'package:marvelapp_flutter/presentation/screens/details_screen/bloc/details_screen_bloc.dart';
+import 'package:marvelapp_flutter/presentation/screens/details_screen/bloc/details_screen_state.dart';
+import 'package:marvelapp_flutter/presentation/widgets/details_content.dart';
+import '../../../data/model/response_models/character.dart';
+import '../../../data/model/response_models/series.dart';
+import '../../widgets/custom_error_widget.dart';
+import 'bloc/details_screen_event.dart';
+
+class DetailsScreen extends StatelessWidget {
+  DetailsScreen({Key? key}) : super(key: key);
+  final MarvellRepository repository = DioMarvellRepository();
+
+  @override
+  Widget build(BuildContext context) {
+    final characterId = ModalRoute.of(context)!.settings.arguments.toString();
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("MarvellApp - Detail"),
+      ),
+      body: BlocProvider(
+        create: (_) => DetailsScreenBloc(repository: repository, characterId: characterId)..add(GetCharacterDetail()),
+        child: BlocBuilder<DetailsScreenBloc, DetailsScreenState>(
+          builder: (context, state) {
+            Widget child = const Center(
+              child: CircularProgressIndicator(),
+            );
+            if (!state.loading) {
```

Comment:
Insted of the nesting you can just `if (loading) return CircularProgressIndicator();` at the start.

Reply:
done. and add it on home_screen. but add check for next if state.loading